tests/sagemaker-tests/test_sagemaker_utils.py

```diff
@@ -221,3 +233,156 @@ def test_wrap_model_artifacts():
+    for invalid_name in invalid_names:
+        with pytest.raises(ArgumentTypeError, match='failed to satisfy regular expression pattern'):
+            check_name_format(invalid_name)
+
+@patch('launch_utils.boto3.client')
+def test_check_tarfile_s3_object(mock_boto_client):
+    """ The the check if tarfile object in S3 url correct and exist
+    """
+    mock_s3 = Mock()
+    mock_boto_client.return_value = mock_s3
+
+    # Test case 1: normal case. S3 url is right and the object ends with '.tar.gz' 
+    mock_s3.head_object.return_value = "s3://graphstorm-demos/test/model.tar.gz"
+    assert check_tarfile_s3_object("s3://graphstorm-demos/test/model.tar.gz") is True
+
+    # Test case 2: abnormal cases
+    #       2.1: S3 url is incorrect, not starting with s3 or https
+    error_response = {'Error': {'Code': '404'}}
+    mock_s3.head_object.side_effect = ClientError(error_response, 'HeadObject')
+    with pytest.raises(AssertionError, match='Incorrect S3'):
+        check_tarfile_s3_object('/graphstorm-demos/test/GSF-demo.pptx')
+
+    #       2.2: S3 url is correct, but not ending with .tar.gz
+    error_response = {'Error': {'Code': '404'}}
+    mock_s3.head_object.side_effect = ClientError(error_response, 'HeadObject')
+    with pytest.raises(AssertionError, match='not a compressed tar file'):
+        check_tarfile_s3_object('s3://graphstorm-demos/test/GSF-demo.pptx')
+
+@patch('launch_utils.S3Uploader.upload')
+def test_upload_data_to_s3(mock_s3uploader):
+    """ Test the upload data to S3 function.
+    """
+    mock_s3uploader.return_value = 's3://graphstorm-demos/test/model.tar.gz'
+    
+    # Test case 1: mock successful upload
+    ret = upload_data_to_s3('s3://graphstorm-demos/test/', './model.tar.gz', 'session')
+    mock_s3uploader.assert_called_once_with('./model.tar.gz', 's3://graphstorm-demos/test/',
+                                            sagemaker_session='session')
+
+    assert ret == 's3://graphstorm-demos/test/model.tar.gz'
+
+    # Test case 2: mock unsucessful upload
+    ret = upload_data_to_s3('s3://graphstorm-demos/test/', './model.tar.gz', 'session')
+    with pytest.raises(AssertionError):
+        mock_s3uploader.assert_called_once_with('s3://graphstorm-demos/test/', './model.tar.gz',
+                                                sagemaker_session='session')
+    
+
+def local_test_check_tarfile_s3_object(input_value, expected_result):
+    """ Test the check if a model tarfile exist in the given S3 url location.
+
+    ## This test should NOT be ran in CI pipeline, but in local environment only! ##
+    
+    Testers need to provide an S3 url as the input value, and True or False as the
+    expected result in the python command line.
+
+    """
+    s3_url = input_value
+
+    if expected_result.lower() == 'true':
+        assert check_tarfile_s3_object(s3_url)
+    else:
+        assert not check_tarfile_s3_object(s3_url)
+
+
+def local_test_upload_data_to_s3(input_value, expected_result=None):
```

Comment:
This function is not called by pytest.

Reply:
No. Because our CI pipeline may not be able to access S3, neither has an S3 object in test code, this test case is for local running only as the main function defined.

Reply:
If it is not called by pytest, we may not want to keep it here.
@thvasilo what is your opinion?